capstone/src/test/java/com/google/sps/servlets/DataServletTest.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+import com.google.sps.servlets.DataServlet;
+import com.google.sps.data.MockPerspectiveCaller;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+/** Test the DataServlet class*/
+@RunWith(JUnit4.class)
+public final class DataServletTest {
+  private DataServlet dataServlet;
+
+  @Before
+  public void setUp() {
+    MockPerspectiveCaller mock = new MockPerspectiveCaller();
+    this.dataServlet = new DataServlet(mock);
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+    HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+  }
+
+  @Test
+  public void PerspectiveCallWithNullInput() throws IOException{
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(writer);
+
+    dataServlet.doGet(request, response);
+
+    Assert.assertTrue(stringWriter.toString().contains("{\"attributeScores\":{"));
```

Comment:
Open to suggestionson how to better check the output of the call; this simply verifies that the object returned has an attributeScores key.

Reply:
I think this is good